File: dotnet/src/Microsoft.Agents.Protocol.Validation/Validators/ContentValidators/ErrorContentValidator.cs
using Microsoft.Agents;

namespace Microsoft.Agents.Protocol.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for ErrorContent.
/// </summary>
public class ErrorContentValidator : ContentValidatorBase<ErrorContent>
{
    public override ValidationResult Validate(ErrorContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-011: ErrorContent message must be non-empty
        var messageError = ValidateNotEmpty(content.Message, "Message",
            ValidationErrorCode.CNT_011,
            "ErrorContent message must be non-empty");
        if (messageError != null)
            errors.Add(messageError);

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/ErrorContentValidator.ts
import type { ErrorContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';
import { ValidationErrorCode } from '../../core/ValidationErrorCode';

/**
 * Validator for ErrorContent.
 */
export class ErrorContentValidator extends ContentValidatorBase<ErrorContent> {
  public validate(content: ErrorContent, context?: ValidationContext): ValidationResult {
    const errors = this.collectErrors(
      // CNT-011: error message must be non-empty
      this.validateNotEmpty(
        content.message,
        'message',
        ValidationErrorCode.CNT_011,
        'ErrorContent message must be non-empty'
      )
    );

    return new ValidationResult(errors);
  }
}
